### PR TITLE
state: keep a cache of block verification results

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: #1032

### CODE REVIEW

File: state/execution.go [hunk 1]
```diff
@@ -37,9 +38,11 @@ type BlockExecutor struct {
 	mempool mempl.Mempool
 	evpool  EvidencePool
 
-	logger log.Logger
-
+	logger  log.Logger
 	metrics *Metrics
+
+	// cache the verification results over a single height
+	cache []bytes.HexBytes
```
Comment (alexanderbez) [inline state/execution.go:45]: why not consider `map[string]struct{}`? Granted, there is overhead for `hash.String()`, but it's cleaner and possibly more performant and an iterative search.

Comment (cmwaters) [reply state/execution.go:45]: Yeah I had considered it was pretty indifferent between the two. I'll make the change

**Review by alexanderbez — APPROVED** [review]

File: state/execution.go
Comment (Walaszczyk) [inline state/execution.go:1]: state/execution.go

**Review by Walaszczyk — COMMENTED** [review]
gh pr checkout 64026402callum/cache-verificationhttps://github.com/tendermint/tendermint/pull/5610#discussion_r1938575677https://github.com/tendermint/tendermint/pull/5610#discussion_r516454567

**Pull request merged**